### PR TITLE
SA-1680: Parse query param for query_filters

### PR DESCRIPTION
**Pull request by PatrickSison**
### What Changed
 - Parses for `query_filters` and returns with helper function
 - Updates `query_filters` in URL query params

### How To Test
 - Add console.log to parseSearchNew to check `ret.queryFilters`
 - Use param `?query_filters=%7B%22groupings%22%3A%5B%7B%22AND%22%3A%7B%22domains%22%3A%7B%22eq%22%3A%5B%22gmail.com%22%2C%22yahoo.com%22%2C%22hotmail.com%22%5D%2C%22like%22%3A%5B%22mail%22%5D%7D%2C%22sending_domains%22%3A%7B%22not%22%3A%7B%22eq%22%3A%5B%22sparkpost.com%22%5D%7D%7D%7D%7D%2C%7B%22OR%22%3A%7B%22templates%22%3A%7B%22eq%22%3A%5B%22default%22%5D%7D%2C%22campaigns%22%3A%7B%22like%22%3A%5B%22333%22%2C%22344%22%2C%22355%22%5D%7D%7D%7D%2C%7B%22OR%22%3A%7B%22campaigns%22%3A%7B%22not%22%3A%7B%22like%22%3A%5B%22Friday%22%5D%2C%22eq%22%3A%5B%22stuff%22%2C%22things%22%5D%7D%7D%7D%7D%5D%7D`
 - Verify that object is parsed correctly

### To Do
- [ ] Swap out parseSearch for parseSearchNew where used in report builder

### Notes
- A bit dependent on SA-1664 to have new storage available.

### CODE REVIEW

**Review by shrirupa — APPROVED** [review]
Looks good to me!

File: src/helpers/reports.js [hunk 1]
```diff
@@ -74,3 +74,90 @@ export function parseSearch(search) {
...
+  'Recipient Domain': 'domains',
+  Campaign: 'campaigns',
+  Template: 'templates',
+  'Sending IP': 'sending_ips',
+  'IP Pool': 'ip_pools',
+  Subaccount: 'subaccounts',
+  'Sending Domain': 'sending_domains',
+};
+
+function mapFiltersToComparators(filters) {
```
Comment (PatrickSison) [inline src/helpers/reports.js:155]: Old filters are converted to new filters, regardless of feature flag (needs to support after feature flag removed)

File: src/pages/reportBuilder/components/ReportOptions.js [hunk 2]
```diff
@@ -90,9 +75,22 @@ export function ReportOptions(props) {
   // Updates the query params with incoming search option changes
   useEffect(() => {
     if (reportOptions.isReady) {
-      updateRoute(selectSummaryChartSearchOptions);
+      const { filters, ...update } = selectSummaryChartSearchOptions;
+      if (isComparatorsEnabled) {
+        update.query_filters = JSON.stringify(filters);
```
Comment (PatrickSison) [inline src/pages/reportBuilder/components/ReportOptions.js:80]: Both cases stored as filters, but changed to query_filters if it's new structure. Will remove 2nd block as we transition off of the filters variable

File: src/pages/reportBuilder/context/ReportBuilderContext.js [hunk 3]
```diff
@@ -100,6 +103,62 @@ const reducer = (state, action) => {
...
+      if (!update.timezone || !useMetricsRollup) {
+        update.timezone = getLocalTimezone();
+      }
+
+      if (!update.metrics) {
+        update.metrics = config.reportBuilder.defaultMetrics;
+      }
+
+      // If queryFilters, then was converted over from older filters.
+      if (payload.queryFilters) {
```
Comment (PatrickSison) [inline src/pages/reportBuilder/context/ReportBuilderContext.js:123]: If has queryFilters (which only comes from parsing the URL params), sets filters as that variable.

**Review by shrirupa — APPROVED** [review]

**Pull request merged**